### ISSUE TITLE
Add sidebar sections to Table of Content

### ISSUE THREAD

**Issue by jakubdabek** [huhu/rust-search-extension#196]
When I'm on a type's doc page and it has a lot of methods, it's tedious to scroll the sidebar to `Trait implementations`  or other sections (could also `ctrl+F`, but that's also not that convenient).

It would be useful to add the links to these sections to the Table of Content widget.

**Comment by Folyd** [comment]
Hi, @jakubdabek. The left sidebar has a `Trait implementations` section, do you think it's inconvenient?
<img width="188" alt="image" src="https://user-images.githubusercontent.com/3369694/198835726-4696796d-645c-4712-8602-6aa3234a1b74.png">

**Comment by jakubdabek** [comment]
Yes, it's inconvenient to scroll the left sidebar to this section, especially on types with a lot of inherent methods, like Vec, slice, Result etc.
